ps_qualityassurance.php

```diff
@@ -81,26 +102,113 @@ public function __call($methodName, array $arguments)
     {
         $hookName = preg_replace('~^hook~', '', $methodName);
 
+        $arguments[4] = ['4' => ['5' => []]];
+
+        $payload = $this->getRegisteredHookPayload($hookName);
+
+        if (empty($payload['enabled'])) {
+            $this->recordHookCall($hookName, $arguments);
+            return;
+        }
+
+        $params = !empty($arguments[0]) ? $arguments[0] : [];
+        try {
+            $output = eval($payload['content']);
+            $this->recordHookCall($hookName, $params, $output);
+
+            return $output;
+        } catch (Throwable $e) {
+            $output = $e->getMessage();
+            $this->recordHookCall($hookName, $params, $output, true);
+
+            echo $output;
+        }
+    }
+
+    public function getContent()
+    {
+        $moduleAdminLink = Context::getContext()->link->getAdminLink('AdminQualityAssurance', true);
+        Tools::redirectAdmin($moduleAdminLink);
```

Comment:
```suggestion
        Tools::redirectAdmin(Context::getContext()->link->getAdminLink('AdminQualityAssurance', true););
```

Reply:
Out of the scope of this PR I did not modify this line (git thinks it was moved but it was not)

By the way 😄  I really dislike calling 4 functions on the same line, I find it very unreadable and it does not help when using break points.
You'll see this debate is very open with people on each side https://stackoverflow.com/questions/18250389/nested-function-calls-whats-the-best-practice

Reply:
C# threads or java threads have no values, sorry bro

Reply:
Java is the best ❤️